lib/pgBackRest/Config/Config.pm

```diff
@@ -2793,34 +682,30 @@ sub optionValidate
                 }
 
                 # Process an allow list for the command then for the option
-                my $oAllow = defined($oCommandRule) ? $$oCommandRule{&OPTION_RULE_ALLOW_LIST} :
-                                                      $oOptionRule{$strOption}{&OPTION_RULE_ALLOW_LIST};
-
-                if (defined($oAllow) && !defined($$oAllow{$strValue}))
+                if (cfgRuleOptionAllowList($iCommandId, $iOptionId) &&
+                    !cfgRuleOptionAllowListValueValid($iCommandId, $iOptionId, $strValue))
                 {
                     confess &log(ERROR, "'${strValue}' is not valid for '${strOption}' option", ERROR_OPTION_INVALID_VALUE);
                 }
 
                 # Process an allow range for the command then for the option
-                $oAllow = defined($oCommandRule) ? $$oCommandRule{&OPTION_RULE_ALLOW_RANGE} :
-                                                     $oOptionRule{$strOption}{&OPTION_RULE_ALLOW_RANGE};
-
-                if (defined($oAllow) && ($strValue < $$oAllow[0] || $strValue > $$oAllow[1]))
+                if (cfgRuleOptionAllowRange($iCommandId, $iOptionId) &&
```

Comment:
These are all instances where a command *can* override rules but doesn't currently, or the pattern of overrides allows the switch logic to optimize away command id.  For instance, cfgRuleOptionDepend uses command id but cfgRuleOptionDependOption doesn't because it has been optimized away.